File: truptipatil/Microsoft/Day1/EvaluateReversePolishNotation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<long long int>s;

        for(long long int i=0;i<tokens.size();i++){
            if(tokens[i]!="+" && tokens[i]!="-" && tokens[i]!="*" && tokens[i]!="/"){
                s.push(stoi(tokens[i]));
            }else{
                long long int a=s.top();
                s.pop();

                if(tokens[i]=="+")
                    s.top()+=a;
                else if(tokens[i]=="-")
                    s.top()-=a;
                else if(tokens[i]=="*")
                    s.top()*=a;
                else if(tokens[i]=="/")
                    s.top()/=a;
            }
        }
        return s.top();
    }
};

File: Calto-GG/Microsoft/Day 1/Evaluate_Reverse_Polish_Notation.py
class Solution:
    def evalRPN(self, tokens: list[str]) -> int:
        stack=[]
        for i in tokens:
            if i in '+-*/': a,b=stack.pop(),stack.pop()
            if i == '+': stack.append(b+a)
            elif i == '-': stack.append(b-a)
            elif i == '*': stack.append(b * a)
            elif i == '/': stack.append(int(b/a))
            else: stack.append(int(i))
        return stack[0]